File: back-end/src/main/java/com/huyphan/mappers/NewChatMessageDataMapper.java
package com.huyphan.mappers;

import com.huyphan.dtos.NewChatMessageDataDto;
import com.huyphan.mappers.converters.StringToInstantConverter;
import com.huyphan.models.NewChatMessageData;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;


@Component
public class NewChatMessageDataMapper extends BaseMapper implements
        FromDtoMapper<NewChatMessageDataDto, NewChatMessageData> {

    @Autowired
    private StringToInstantConverter converter;

    @Override
    public NewChatMessageData fromDto(NewChatMessageDataDto data) {
        return modelMapper.map(data, NewChatMessageData.class);
    }

    @Override
    public void createTypeMap() {
        modelMapper.createTypeMap(NewChatMessageDataDto.class, NewChatMessageData.class)
                .addMappings(
                        mapper -> mapper.using(converter)
                                .map(NewChatMessageDataDto::getSentDate,
                                        NewChatMessageData::setSentDate)
                );
    }
}

File: client/src/services/mappers/new-chat-message-data-mapper.ts
import NewChatMessageData from '../../models/new-chat-message-data';
import NewChatMessageDataDto from '../dtos/new-chat-message-data-dto';
import { MessageContentMapper } from './message-content-mapper';

export namespace NewChatMessageDataMapper {
  export function toDto(data: NewChatMessageData): NewChatMessageDataDto {
    return {
      ...data,
      content: MessageContentMapper.toDto(data.content),
      sentDate: data.sentDate.toISOString(),
    };
  }
}
